mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -281,16 +280,17 @@ class Word2Vec extends Serializable with Logging {
     val expTable = sc.broadcast(createExpTable())
     val bcVocab = sc.broadcast(vocab)
     val bcVocabHash = sc.broadcast(vocabHash)
-
-    val sentences: RDD[Array[Int]] = words.mapPartitions { iter =>
+    //each partition is a collection of sentences, will be translated into arrays of Index integer
+    val sentences: RDD[Array[Int]] = dataset.mapPartitions { iter =>
```

Comment:
Agree that Word2Vec can be used in settings where the "sentence" is just a sequence of "things" (e.g. recommendation where the "sentences" could be "page views" for a user, and could be longer than 1000).

So while I'd tend to agree that the caller should really be responsible for setting up the "sentence" structure, as @srowen says rather make it configurable with a default of `1000` to keep backward compatible behavior. You can create a `set` method for the new parameter and add a `@Since` annotation (see e.g. the `setMinCount` method added)